dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapConnection.java

```diff
@@ -2073,38 +2078,41 @@ private void supportRangeAttributes(Attributes attributes, String dn, TimedDirCo
                     Tr.debug(tc, METHODNAME + " Range attribute retrieved: " + attrName);
                 }
 
+                //Add the ranged attributes we found to newAttr
                 for (NamingEnumeration<?> neu2 = attr.getAll(); neu2.hasMoreElements();) {
                     newAttr.add(neu2.nextElement());
                 }
 
-                int rangeLow = iAttrRangeStep;
+                //Parse the range attribute for low and high values to make sure
+                //we are using the maximum step range.
+                int localAttrRangeStep = parseRangeAttribute(attrName);
+
+                //The first set of attributes has been added. Now update the range
+                //and retrieve the second set.
+                int rangeLow = localAttrRangeStep;
                 try {
                     rangeLow = Integer.parseInt(attrName.substring(attrName.indexOf('-') + 1)) + 1;
                 } catch (NumberFormatException e) {
                     // continue;
                 }
-                int rangeHigh = rangeLow + iAttrRangeStep - 1;
+                int rangeHigh = rangeLow + localAttrRangeStep - 1;
                 boolean quitLoop = false;
-                boolean lastQuery = false;
                 do {
+                    //Format the getAttributes request
                     String attributeWithRange = null;
-                    if (!lastQuery) {
-                        Object[] args = { Integer.valueOf(rangeLow).toString(),
-                                          Integer.valueOf(rangeHigh).toString()
-                        };
-                        attributeWithRange = attrId + MessageFormat.format(ATTR_RANGE_QUERY, args);
-                    } else {
-                        Object[] args = {
-                                          Integer.valueOf(rangeLow).toString()
-                        };
-                        attributeWithRange = attrId + MessageFormat.format(ATTR_RANGE_LAST_QUERY, args);
-                    }
+                    Object[] args = {
+                                      new Integer(rangeLow).toString(),
```

Comment:
This would be more efficient:

```suggestion
                                      Integer.toString(rangeLow),
```